src/storage/src/hummock/event_handler/hummock_event_handler.rs

```diff
@@ -518,31 +521,61 @@ impl HummockEventHandler {
     }
 
     fn resolve_version_update_info(
-        pinned_version: PinnedVersion,
+        pinned_version: &PinnedVersion,
         version_payload: HummockVersionUpdate,
         mut sst_delta_infos: Option<&mut Vec<SstDeltaInfo>>,
     ) -> Option<PinnedVersion> {
-        let newly_pinned_version = match version_payload {
+        match version_payload {
             HummockVersionUpdate::VersionDeltas(version_deltas) => {
-                let mut version_to_apply = (*pinned_version).clone();
-                for version_delta in &version_deltas {
-                    assert_eq!(version_to_apply.id, version_delta.prev_id);
-                    if let Some(sst_delta_infos) = &mut sst_delta_infos {
-                        sst_delta_infos.extend(
-                            version_to_apply
-                                .build_sst_delta_infos(version_delta)
-                                .into_iter(),
+                let mut version_to_apply = pinned_version.deref().clone();
+                {
+                    let mut table_change_log_to_apply_guard =
+                        pinned_version.table_change_log_write_lock();
+                    // let mut table_change_log_to_apply_guard = table_change_log_to_apply.write();
+                    for version_delta in &version_deltas {
```

Comment:
```suggestion
                    for version_delta in version_deltas {
```

And then below we use do 
```
let local_hummock_version_delta = LocalHummockVersionDelta::from(version_delta);
```